umap/views.py

```diff
@@ -1033,42 +1032,41 @@ class DataLayerCreate(FormLessEditMixin, GZipMixin, CreateView):
     def form_valid(self, form):
         form.instance.map = self.kwargs["map_inst"]
         self.object = form.save()
-        # Simple response with only metadatas (including new id)
+        # Simple response with only metadata (including new id)
         response = simple_json_response(
             **self.object.metadata(self.request.user, self.request)
         )
-        response["Last-Modified"] = self.last_modified
+        response["X-Umap-Version"] = self.version
         return response
 
 
 class DataLayerUpdate(FormLessEditMixin, GZipMixin, UpdateView):
     model = DataLayer
     form_class = DataLayerForm
 
-    def has_been_modified_since(self, if_unmodified_since):
-        return if_unmodified_since and self.last_modified != if_unmodified_since
+    def has_changes_since(self, incoming_version):
```

Comment:
Given we are using version number and not time anymore, maybe rename to `match_version`  or `is_latest`/`is_latest_version`?

Reply:
I prefer `has_changes_since` because we're passing the argument with the version. With the propositions you're making, it's less clear to me that it's supposed to be an argument passed.